### PR TITLE
[rush-init] update template pnpm to latest version

### PR DESCRIPTION
**Pull request by chengcyber**
## Details

when `rush init` the default pnpm version is 4, this PR try to update to use the latest version of pnpm

### CODE REVIEW

File: common/changes/@microsoft/rush/update-init-template_2021-01-08-02-53.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "packageName": "@microsoft/rush",
+      "comment": "use pnpm 5 when rush init",
```
Comment (iclanton) [inline common/changes/@microsoft/rush/update-init-template_2021-01-08-02-53.json:5]:
```suggestion
      "comment": "Update rush.json produced by rush init to use PNPM 5.14.3",
```

**Comment by iclanton** [comment]
Have you tested this version of PNPM in a moderately large repo? @halfnibble is going to try this out in one of our internal repos.

**Comment by chengcyber** [comment]
> Have you tested this version of PNPM in a moderately large repo? @halfnibble is going to try this out in one of our internal repos.

I am migrating a monorepo project to use `rush`, with `pnpm@5`. All good for now.

And, yes. I think it's worth to do fully test before merge, since my project doesn't cover every cases.

**Review by iclanton — APPROVED** [review]

**Comment by octogonz** [comment]
This was released with Rush 5.37.0

**Pull request merged**